### ISSUE TITLE
Kanopy site support?

### ISSUE THREAD

**Issue by Winkster** [ytdl-org/youtube-dl#20423]
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2019.03.18*. If it's not, read [this FAQ entry](https://github.com/ytdl-org/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2019.03.18**

### Before submitting an *issue* make sure you have:
- [x] At least skimmed through the [README](https://github.com/ytdl-org/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/ytdl-org/youtube-dl#faq) and [BUGS](https://github.com/ytdl-org/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones
- [x] Checked that provided video/audio/playlist URLs (if any) are alive and playable in a browser

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
C:\Users\****\Downloads>youtube-dl -v --cookies kanopy.cookies.txt https://nypl.kanopy.com/video/ken-burns-empire-air
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', '--cookies', 'kanopy.cookies.txt', 'https://nypl.kanopy.com/video/ken-burns-empire-air']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2019.03.18
[debug] Python version 3.4.4 (CPython) - Windows-10-10.0.17763
[debug] exe versions: ffmpeg 4.1.1, ffprobe 4.1.1
[debug] Proxy map: {}
[generic] ken-burns-empire-air: Requesting header
WARNING: Falling back on generic information extractor.
[generic] ken-burns-empire-air: Downloading webpage
[generic] ken-burns-empire-air: Extracting information
ERROR: Unsupported URL: https://nypl.kanopy.com/video/ken-burns-empire-air
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpcs52imf5\build\youtube_dl\YoutubeDL.py", line 794, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpcs52imf5\build\youtube_dl\extractor\common.py", line 529, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmpcs52imf5\build\youtube_dl\extractor\generic.py", line 3320, in _real_extract
youtube_dl.utils.UnsupportedError: Unsupported URL: https://nypl.kanopy.com/video/ken-burns-empire-air

<end of log>
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):

- Single video: https://nypl.kanopy.com/video/battleship-potemkin [PUBLIC DOMAIN, see description below]
- Playlist: https://nypl.kanopy.com/playlist/5597146

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

[Kanopy](https://kanopy.com) is [very popular with libraries in the US](https://www.nytimes.com/2017/08/24/watching/kanopy-criterion-collection-library-card.html), most notably the NYPL. You should check to see if [your local library](https://www.kanopy.com/wayf) is supported. Anyway, Kanopy is a treasure trove of educational and/or [public domain films](https://en.wikipedia.org/wiki/Public_domain_film#United_States) and would be an excellent supported site addition to this already excellent app. THANK YOU!

**Comment by quinlander** [comment]
This needs label "account-needed"

**Comment by adriel** [comment]
I don't think it'll be possible, looks like they use widevine DRM.
![image](https://user-images.githubusercontent.com/154217/79924932-3b65bd80-848d-11ea-9529-2e1e07506b24.png)

**Comment by Winkster** [comment]
Bummer, but thanks so much for investigating!

A pity, as Kanopy is a great service, and the sorts of collections
libraries serve up through them are in many cases long out of print or hard
to find. Thanks again for the 411

**Comment by aereaux** [comment]
I think that not all the videos have DRM, mostly with the documentaries.  For example this one (https://www.kanopy.com/product/race) seems to work in firefox without having enabled DRM.  Would it be possible to have support for these?

**Comment by 89z** [comment]
I think I can implement this, if someone can share a login

**Comment by 89z** [comment]
https://github.com/ytdl-org/youtube-dl/issues/28887

**Comment by garymm** [comment]
@89z I just shared an account with you

**Comment by 89z** [comment]
Yes I see, thanks. I am busy implementing Apple TV+ at this time, but I suppose I can take a break to look at Kanopy.

**Comment by 89z** [comment]
I was able to successfully download and decrypt this video:

https://kanopy.com/product/113763

using method described here:

https://github.com/89z/mech/tree/481b33ff/research/kanopy

my notes are just a proof of concept. I am not really interested in the site personally, so I will only implement a tool if someone contracts me to do it. If you want to do that, check my bio for contact, otherwise good luck!

**Comment by CHJ85** [comment]
@adriel I've downloaded many videos from there using Video DownloadHelper.
And those are not encrypted with Widevine. However, a lot of videos are. It's a hit or miss, like Tubi.
However, I'd like to +1 this request.

**Comment by 101Dude** [comment]
What worked for me with Kanopy was to open the m3u8 in another window and run:

`yt-dlp --cookies-from-browser safari --list-formats` 

I haven't tested that on many videos...  it worked on one.

**Comment by SirMikeDouglas** [comment]
+1

**Comment by dirkf** [comment]
This site is a Nuxt-based JS app. Apparently media data is only revealed after logging in to an affiliated university site or public library.

Someone who wants to work on this will need a suitable account, theirs or borrowed. Or someone with such an account might make a HAR trace of a session where a video is played using the browser's dev tools, for a developer to analyse; but please ask first since such a trace will be large and contain sensitive data.
